tests/std/tests/GH_002030_asan_annotate_string/test.cpp

```diff
@@ -170,29 +166,23 @@ class input_iterator_tester {
 };
 
 template <class CharType, class Alloc>
-bool verify_string(basic_string<CharType, char_traits<CharType>, Alloc>& str) {
+bool verify_string(const basic_string<CharType, char_traits<CharType>, Alloc>& str) {
 #ifdef __SANITIZE_ADDRESS__
-    constexpr auto proxy_size = _Size_after_ebco_v<_Container_base>;
-    if constexpr (proxy_size % _Asan_granularity != 0) { // If we have a misaligned SSO buffer we disable ASAN
-        constexpr size_t max_sso_size = (16 / sizeof(CharType) < 1 ? 1 : 16 / sizeof(CharType)) - 1;
-        if (str.capacity() == max_sso_size) {
-            return true;
-        }
-    }
+    const void* const buffer  = str.data();
+    const void* const buf_end = str.data() + (str.capacity() + 1);

```

Comment:
Suggestion: The parentheses could be omitted here, which would be consistent with `str.data() + str.size() + 1` below.